### PR TITLE
Add gometalinter errcheck to pre-commit-hook, fix all Go unchecked errors.

### PR DESCRIPTION
**Pull request by helinwang**

### CODE REVIEW

File: paddle/scripts/travis/check_style.sh [hunk 1]
```diff
@@ -13,6 +13,12 @@ export PATH=/usr/bin:$PATH
 pre-commit install
 clang-format --version
 
+# set up go environment for running gometalinter
```
Comment (wangkuiyi) [inline paddle/scripts/travis/check_style.sh]: Who would call check_style.sh?

Comment (wangkuiyi) [inline paddle/scripts/travis/check_style.sh]: Oh. I see. I recommended to call check_style.sh from Travis CI.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -21,10 +21,12 @@
...
     -   id: clang-formater
--   repo: https://github.com/dnephin/pre-commit-golang
-    sha: e4693a4c282b4fc878eda172a929f7a6508e7d16
+-   repo: https://github.com/PaddlePaddle/pre-commit-golang
+    sha: 9cbbd1642aef9a209bd7f49dddf1ca20cd20c4d3
     hooks:
       -   id: go-fmt
-          files: (.*\.go)
+          types: [go]
       -   id: go-lint
```
Comment (gongweibao) [inline .pre-commit-config.yaml]: 用了`gometalinter`应该就不用再单独使用`golint`了。
https://github.com/alecthomas/gometalinter#quickstart

File: paddle/scripts/travis/check_style.sh [hunk 3]
```diff
@@ -13,6 +13,16 @@ export PATH=/usr/bin:$PATH
...
+mkdir -p $GOPATH/src/github.com/PaddlePaddle/
+ln -sf $TRAVIS_BUILD_DIR $GOPATH/src/github.com/PaddlePaddle/Paddle
+cd  $GOPATH/src/github.com/PaddlePaddle/Paddle/go; glide install; cd -
+
+echo "gopath: $GOPATH"
+echo "pwd: `pwd`"
+find ./go
```
Comment (gongweibao) [inline paddle/scripts/travis/check_style.sh]: 这个是做什么？是测试代码？

File: paddle/scripts/travis/check_style.sh [hunk 4]
```diff
@@ -13,6 +13,16 @@ export PATH=/usr/bin:$PATH
...
+cd go; gometalinter --vendor --disable-all --enable errcheck ./...; cd -
```
Comment (gongweibao) [inline paddle/scripts/travis/check_style.sh]: `disable-all linters`的原因是？

File: go/pserver/service.go [hunk 5]
```diff
@@ -219,7 +219,7 @@ func (s *Service) GetParam(name string, parameter *Parameter) error {
 }
 
 // pserver save checkpoint
-func (s *Service) doCheckpoint() error {
+func (s *Service) doCheckpoint() (err error) {
```
Comment (dzhwinter) [inline go/pserver/service.go]: a smart feature of named return value!
seems that I ignored so much return error when saving the checkpoint.

File: paddle/scripts/travis/check_style.sh (hunk 4 above)
Comment (gongweibao) [inline paddle/scripts/travis/check_style.sh]: 我们是否也把`vet`检查加上？

File: paddle/scripts/travis/check_style.sh (hunk 3 above)
Comment (dzhwinter) [inline paddle/scripts/travis/check_style.sh]: maybe @helinwang wants to listing the directory to log file. That's reasonable.

Comment (dzhwinter) [inline paddle/scripts/travis/check_style.sh]: we need to filter out the vendor package file. change to
```
find ./go  -not -path "./go/vendor/*"
```

Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: @gongweibao @dzhwinter 不好意思，是测试的，不知为何在我本地能运行，在travis不行，加了写代码测试了一下。

File: paddle/scripts/travis/check_style.sh (hunk 4 above)
Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: @gongweibao `go vet` 有一些正确的地方过不了。

File: paddle/scripts/travis/check_style.sh (hunk 3 above)
Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: @dzhwinter good point! Please see https://github.com/PaddlePaddle/pre-commit-golang/blob/master/run-gometalinter.sh , `--vendor` already ignore the vendor folder.

Comment (dzhwinter) [inline paddle/scripts/travis/check_style.sh]: get it.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (helinwang) [inline .pre-commit-config.yaml]: @gongweibao done.

File: paddle/scripts/travis/check_style.sh (hunk 3 above)
Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: @gongweibao @dzhwinter all removed, found the issue:
go tool的对象代码（比如go build xxx）最好是在$GOPATH里面，不然会有各种问题（vendor directory ignored, no caching of built object files...)。我是调travis gometalinter的时候出现的这个问题，如果运行对象不在$GOPATH中，gometalinter一直过不了。改成了这样才过的：https://github.com/PaddlePaddle/pre-commit-golang/blob/master/run-gometalinter.sh#L5

File: paddle/scripts/travis/check_style.sh (hunk 4 above)
Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: Here is the message showing `go vet` fails:
```
gometalinter --vendor --disable-all --enable errcheck --enable vet --enable golint ./...
master/c/client.go:26::error: possible misuse of unsafe.Pointer (vet)
pserver/optimizer.go:17::error: possible misuse of unsafe.Pointer (vet)
pserver/client/c/cclient.go:37::error: possible misuse of unsafe.Pointer (vet)
```

Comment (helinwang) [inline paddle/scripts/travis/check_style.sh]: @gongweibao , 修复了，其实`go vet`都报对了。

Comment (gongweibao) [inline paddle/scripts/travis/check_style.sh]: 👍

**Pull request merged**